text/0000-install-pin-npm.md

```diff
@@ -0,0 +1,65 @@
+- Feature Name: install_pin_npm
+- Start Date: 2019-10-14
+- RFC PR: (leave this empty)
+- Volta Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Support installation and pinning of arbitrary `npm` versions, decoupled from the selected Node version.
+
+# Motivation
+[motivation]: #motivation
+
+For package managers, we currently only support using an arbitrary version of `yarn`, but not `npm`. For `npm`, users are forced to use the version that is automatically bundled with Node, meaning they can't upgrade or downgrade without changing the Node version. We should support `npm` to the same level that we support `yarn` and give users full control over the version of their package manager.
+
+# Pedagogy
+[pedagogy]: #pedagogy
+
+There shouldn't be anything new we need to document / teach users, as the behavior already exists for Node and Yarn. The only change is that we will support the `volta install npm` and `volta pin npm` commands in the same way that we support `volta install yarn` and `volta pin yarn`.
+
+# Details
+[details]: #details
+
+## Installation
+
+Unlike Yarn, `npm` doesn't provide a pre-packaged tarball for download. Instead, the rely on using the existing `npm` installation that is provided with Node to do any upgrades. This is very similar to how we handle global 3rd-party packages, so we should be able to leverage a similar approach for installing `npm`. That does mean, however, that in the same way we require Node to be available to install packages, we will also require Node to be available to install a custom version of `npm`.
+
+It appears we can also skip the `npm install` step, as the `npm` package tarball comes with all of the dependencies already included.
```

Comment:
If we skip the `npm install` step, what are we using the global node installation for?

Reply:
We're using the global node installation (specifically the bundled `npm` in the global node installation) for resolving the version using `npm view` and fetching the tarball using `npm pack`

Given the lack of an `npm install` step, we could potentially take an approach that more closely mimics `yarn` and builds the distro index and download URL directly from `npmjs.com`, with it's own hooks to support changing the download URL (in the same way that Yarn does).